File: Warm-up Challanges (HackerRank)/Sales by Match/SalesbyMatch.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;


class Result {

    /*
     * Complete the 'sockMerchant' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. INTEGER_ARRAY ar
     */

    public static int sockMerchant(int n, List<Integer> ar) {
    // Write your code here
        Map<Integer, Integer> mapSocks = new HashMap<Integer, Integer>();
        Integer result = 0;
        
        //System.out.println("test");
        for(int i = 0; i < ar.size(); i++) {
            if(mapSocks.containsKey(ar.get(i)) && mapSocks.get(ar.get(i)) > 0) {
                mapSocks.put(ar.get(i), mapSocks.get(ar.get(i))-1);
                result++;
            } else {
                mapSocks.put(ar.get(i), 1);
            }
        }
        
        
        return result;
    }

}

public class SalesbyMatch {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> ar = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        int result = Result.sockMerchant(n, ar);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: Warm-up Challanges (HackerRank)/Sales by Match/Sales by Match.js
"use strict";

const fs = require("fs");

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", (inputStdin) => {
  inputString += inputStdin;
});

process.stdin.on("end", (_) => {
  inputString = inputString
    .replace(/\s*$/, "")
    .split("\n")
    .map((str) => str.replace(/\s*$/, ""));

  main();
});

function readLine() {
  return inputString[currentLine++];
}

// Complete the sockMerchant function below.
function sockMerchant(n, ar) {
  let mapSocks = new Map();
  let result = 0;

  for (let i = 0; i < ar.length; i++) {
    if (mapSocks.get(ar[i]) && mapSocks.get(ar[i]) > 0) {
      mapSocks.set(ar[i], mapSocks.get(ar[i]) - 1);
      result++;
    } else {
      mapSocks.set(ar[i], 1);
    }
  }
  return result;
}

function main() {
  const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

  const n = parseInt(readLine(), 10);

  const ar = readLine()
    .split(" ")
    .map((arTemp) => parseInt(arTemp, 10));

  let result = sockMerchant(n, ar);

  ws.write(result + "\n");

  ws.end();
}
